client/src/firebase.js

```diff
@@ -13,8 +13,73 @@ const firebaseConfig = {
 };
 
 const app = initializeApp(firebaseConfig);
-const analytics = getAnalytics(app);
+getAnalytics(app);
 const googleProvider = new GoogleAuthProvider();
 
 export const auth = getAuth(app);
-export { googleProvider}
+
+// Set persistence to keep users logged in for extended periods instead of Firebase' default 1 hour
+setPersistence(auth, browserLocalPersistence)
+  .then(() => {
+    console.log("Firebase persistence set to local storage");
+  })
+  .catch((error) => {
+    console.error("Error setting Firebase persistence:", error);
+  });
+
+let tokenRefreshInterval;
+let lastActivityTime = Date.now();
+const WEEK_IN_MS = 7 * 24 * 60 * 60 * 1000; // 1 week in milliseconds
+
+const updateLastActivity = () => {
+  lastActivityTime = Date.now();
+};
+
+if (typeof window !== 'undefined') {
+  ['mousedown', 'mousemove', 'keypress', 'scroll', 'touchstart', 'click'].forEach(event => {
+    document.addEventListener(event, updateLastActivity, true);
+  });
+}
```

Comment:
You could use `visibilitychange` to detect Tab focus
```suggestion
if (typeof window !== 'undefined') {
  ['mousedown', 'mousemove', 'keypress', 'scroll', 'touchstart', 'click'].forEach(event => {
    document.addEventListener(event, updateLastActivity, true);
  });
  
  document.addEventListener('visibilitychange', () => {
    if (!document.hidden) {
      updateLastActivity();
    }
  });
}
```